armor-write/src/main/java/com/rapid7/armor/store/S3WriteStore.java

```diff
@@ -382,28 +373,32 @@ public void commit(String transaction, ShardId shardId) {
     DistXact status = getCurrentValues(shardId);
     if (status != null)
       status.validateXact(transaction);
-    saveCurrentValues(shardId, new DistXact(transaction, status == null ? null : status.getCurrent()));
-    trackTenant(shardId.getTenant());
-    try {
-      if (status == null || status.getPrevious() == null)
-        return;
-      String toDelete = PathBuilder.buildPath(shardId.shardIdPath(), status.getPrevious());
-      ListObjectsRequest listObjectsRequest = new ListObjectsRequest()
-          .withBucketName(bucket)
-          .withPrefix(toDelete);
-      ObjectListing objectListing = s3Client.listObjects(listObjectsRequest);
-      while (true) {
-        for (S3ObjectSummary objectSummary : objectListing.getObjectSummaries()) {
-          s3Client.deleteObject(bucket, objectSummary.getKey());
-        }
-        if (objectListing.isTruncated()) {
-          objectListing = s3Client.listNextBatchOfObjects(objectListing);
-        } else {
-          break;
+  
+    boolean isArchiving = status != null && doesObjectExist(bucket, PathBuilder.buildPath(shardId.shardIdPath(), status.getCurrent(), ARCHIVING_MARKER));
+    if (!isArchiving) {
+      saveCurrentValues(shardId, new DistXact(transaction, status == null ? null : status.getCurrent()));
```

Comment:
You still want to commit. The delete is totally just an thing to do after successfully committing. If we fail to delete the old transaction we don't fail the transaction. Thus the archiving check should be before we attempt to delete and shouldn't affect the the commit behaivor.